README.md

```diff
@@ -1,189 +1,337 @@
+fft1d.make_plan_distrib(r, r_part, s, s_part, p, p_part)
 ```
-fft1d.destroy_plan(p)
-```
 
-In the distributed API:
+Note the use of `get_num_nodes` to determine the size of the `p` region and
+partition. The task `make_plan_distrib` is a `__demand(__inline)` task that
+internally performs an index launch over the machine to initialize `p`.
+
+> [!IMPORTANT]
+>
+> Like `make_plan`, `make_plan_distrib` overwrites the input and output regions
+> `r` and `s`.
+
+> [!IMPORTANT]
+>
+> In order for the distributed API to work correctly, it is essential that each
+> task in the index launch inside of `make_plan_distrib` is mapped onto a
+> separate node. This ensures that when the region `p` is used later, there is a
+> plan for every node in the machine.
+
+If there are partitions `r_part` and `s_part` that are distributed around the
+machine, one might do the following to execute the plan:
 
+```lua
+__demand(__index_launch)
+for i in r_part.colors do
+  fft1d.execute_plan_task(r_part[i], s_part[i], p)
+end
 ```
+
+Note: `execute_plan` is a `__demand(__inline)` task (as described above). The
+task `execute_plan_task` is simply a wrapper around execute_plan for
+convenience, to avoid needing to define this explicitly.
+
+> [!IMPORTANT]
+>
+> Because execute_plan is a `__demand(__inline)` task, it will never execute on
+> the GPU (unless the parent task is running on the GPU). Therefore, in most
+> cases it is necessary to use `execute_plan_task` if one wants to use the GPU.
+
+> [!IMPORTANT]
+>
+> While `execute_plan_task` may be executed on the GPU, the contents of the `p`
+> region must still be available on the CPU, because the plans must be used by
+> the host-side code to launch the FFT kernels. Therefore, when
+> `execute_plan_task` is mapped onto the GPU, it is very important to map the
+> `p` region into zero-copy memory.
+
+Lastly, to destroy the plan:
+
+```lua
 fft1d.destroy_plan_distrib(p, p_part)
 ```
 
-Note: these are both `__demand(__inline)` tasks, and
-`destroy_plan_distrib` will internally perform an index launch to
-destroy the plans on each node.
+> [!NOTE]
+>
+> This is a `__demand(__inline)` task, and `destroy_plan_distrib` will
+> internally perform an index launch to destroy the plans on each node.
 
-**Important:** like `make_plan_distrib`, the index launch issued by
-`destroy_plan_distrib` must be mapped so that each point task runs on
-the node where the plan was originally created.
+> [!IMPORTANT]
+>
+> like `make_plan_distrib`, the index launch issued by `destroy_plan_distrib`
+> must be mapped so that each point task runs on the node where the plan was
+> originally created.
 
-## Caveats
+## Future Developments
 
-  * FFTW's planner is *not thread-safe*, but this isn't currently
-    protected by a lock in the Regent wrapper.
+Next items in the pipeline include distributed transforms across multiple nodes.
+Please submit an issue if there are specific features that may be helpful.
```

Comment:
I'd just go back to the to-do list we had before and not commit roadmap plans into a README.

Reply:
I assume you mean the 'missing features' list above. I added that back in, and thus removed these lines